src/main/java/mSearch/filmlisten/WriteFilmlistJson.java

```diff
@@ -22,112 +22,166 @@
 import com.fasterxml.jackson.core.JsonEncoding;
 import com.fasterxml.jackson.core.JsonFactory;
 import com.fasterxml.jackson.core.JsonGenerator;
-import java.io.File;
-import java.io.FileOutputStream;
-import java.util.Iterator;
-import java.util.zip.ZipEntry;
-import java.util.zip.ZipOutputStream;
+import mSearch.Const;
 import mSearch.daten.DatenFilm;
 import mSearch.daten.ListeFilme;
-import mSearch.Const;
 import mSearch.tool.Log;
 import org.tukaani.xz.LZMA2Options;
 import org.tukaani.xz.XZOutputStream;
 
+import java.io.*;
+import java.nio.ByteBuffer;
+import java.nio.channels.Channels;
+import java.nio.channels.ReadableByteChannel;
+import java.nio.channels.WritableByteChannel;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.nio.file.StandardCopyOption;
+
 public class WriteFilmlistJson {
 
+    private void fastChannelCopy(final ReadableByteChannel src, final WritableByteChannel dest) throws IOException {
+        final ByteBuffer buffer = ByteBuffer.allocateDirect(64 * 1024);
+        while (src.read(buffer) != -1) {
+            buffer.flip();
+            dest.write(buffer);
+            buffer.compact();
+        }
+
+        buffer.flip();
+
+        while (buffer.hasRemaining()) {
+            dest.write(buffer);
+        }
+    }
+
+    protected JsonGenerator getJsonGenerator(OutputStream os) throws IOException {
+        JsonFactory jsonF = new JsonFactory();
+        JsonGenerator jg = jsonF.createGenerator(os, JsonEncoding.UTF8);
+        //jg.useDefaultPrettyPrinter(); // enable indentation just to make debug/testing easier
+
+        return jg;
+    }
+
+    /**
+     * Write film data and compress with LZMA2.
+     *
+     * @param datei      file path
+     * @param listeFilme film data
+     */
+    public void filmlisteSchreibenJsonCompressed(String datei, ListeFilme listeFilme) {
+        final String tempFile = datei + "_temp";
+        filmlisteSchreibenJson(tempFile, listeFilme);
+
+        try {
+            Log.sysLog("Komprimiere Datei: " + datei);
+            if (datei.endsWith(Const.FORMAT_XZ)) {
+                final Path xz = testNativeXz();
+                if (xz != null) {
+                    Process p = new ProcessBuilder(xz.toString(), "-9", tempFile).start();
+                    final int exitCode = p.waitFor();
+                    if (exitCode == 0) {
+                        Files.move(Paths.get(tempFile + ".xz"), Paths.get(datei), StandardCopyOption.REPLACE_EXISTING);
+                    }
+                } else
+                    compressFile(tempFile, datei);
+            }
+
+            Files.deleteIfExists(Paths.get(tempFile));
+        } catch (IOException ex) {
+            ex.printStackTrace();
+            Log.sysLog("Komprimieren fehlgeschlagen");
+        } catch (InterruptedException ex) {
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png 'Severity: MAJOR') Either re-interrupt this method or rethrow the "InterruptedException". [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.com/coding_rules#rule_key=squid%3AS2142)